sql/handler.cc

```diff
@@ -1414,6 +1414,25 @@ int ha_commit_trans(THD *thd, bool all)
     goto err;
   }
 
+  if (rw_trans)
+  {
+    for (Ha_trx_info *hi= ha_info; hi; hi= hi->next())
+    {
+      handlerton *ht= hi->ht();
+      if (! hi->is_trx_read_write())
+        continue;
+      if (ht->flags & HTON_NATIVE_SYS_VERSIONING)
+      {
+        TR_table trt(thd, true);
+        if (trt.update())
```

Comment:
What will happen if this function (ha_commit_trans) will fail after this code? There will be an orphan (which is not a big issue) record in TRT and no other harm?